osu.Game.Resources/Shaders/sh_TriangleBorder.fs

```diff
@@ -13,33 +13,27 @@ layout(std140, set = 0, binding = 0) uniform m_BorderData
 
 layout(location = 0) out vec4 o_Colour;
 
-highp float dstToLine(highp vec2 start, highp vec2 end, highp vec2 pixelPos)
+// Distance to the line which starts at (0.0, 1.0) and ends at (0.5, 0.0)
+highp float dstToSideLine(highp vec2 pixelPos)
 {
-    highp float lineLength = distance(end, start);
-
-    if (lineLength < 0.001)
-        return distance(pixelPos, start);
-
-    highp vec2 a = (end - start) / lineLength;
-    highp vec2 closest = clamp(dot(a, pixelPos - start), 0.0, lineLength) * a + start; // closest point on a line from given position
+    highp vec2 a = vec2(0.4472, -0.8944); // (end - start) / distance(end, start)
```

Comment:
I still don't get how is this a constant when it depends on the coordinate of the pixel...

I'm not sure it's worth obfuscating this (by way of needing to hand calculate these values) for a fairly minor performance increase...

Reply:
Ok so this is basically just the normalisation of the vector (0.5, -1)... This deserves a comment.

Reply:
`a` does not depend on pixel coord, it's used to represent the line itself _(I don't remember the name for it though, may be slope or something similar). But alright, I'll revert this then._
Also the whole pr is a fairly minor performance increase, so it's okay if it's not worth merging at all

Reply:
Once i understood what you were trying to do here, I believe this can be simplified further:

```glsl
// Distance to the line which starts at (0.0, 1.0) and ends at (0.5, 0.0)
highp float dstToSideLine(highp vec2 pixelPos)
{
    // Unit vector in the direction of the line.
    const highp vec2 v = vec2(0.4472, -0.8944);

    // Vector in direction of the pixel.
    highp vec2 r = vec2(pixelPos.x, pixelPos.y - 1.0);

    // Cross product of the two vectors.
    return abs(v.x * r.y - v.y * r.x);
}
```

Reply:
> Once i understood what you were trying to do here, I believe this can be simplified further:
> 
> ```glsl
> // Distance to the line which starts at (0.0, 1.0) and ends at (0.5, 0.0)
> highp float dstToSideLine(highp vec2 pixelPos)
> {
>     // Unit vector in the direction of the line.
>     const highp vec2 v = vec2(0.4472, -0.8944);
> 
>     // Vector in direction of the pixel.
>     highp vec2 r = vec2(pixelPos.x, pixelPos.y - 1.0);
> 
>     // Cross product of the two vectors.
>     return abs(v.x * r.y - v.y * r.x);
> }
> ```

Nice, seems to be even faster (I updated perf comparison)